.local/bin/scripts/projectCreate

```diff
@@ -1,100 +1,189 @@
+  [ "$git_init" == "true" ] && git init $project_name
+}
+
+function cpp_init(){ # pass pname and exe_name
+
+  [[  -z "$project_name" || -z "$exe_name" ]] && help && exit
+
+  mkdir $project_name || exit
+
+  mkdir $project_name/src
+  mkdir $project_name/include
+  mkdir $project_name/bin
+  mkdir $project_name/debug
+  touch $project_name/src/main.cc
+
+  (
+    echo "cmake_minimum_required(VERSION 2.8.9)"
+    echo "project($project_name)"
+    echo "include_directories(include)"
+    echo 'file(GLOB_RECURSE SOURCES "src/*".cc)'
+    echo "add_executable($exe_name \${SOURCES})"
+    echo "install(TARGETS $exe_name DESTINATION /usr/bin)"
+  ) > $project_name/CMakeLists.txt
+
+  (
+    echo "#!/bin/sh"
+    echo 'if [[ -z $1 ]]; then'
+    echo 'mkdir -p bin'
+    echo 'cd bin'
+    echo 'cmake ..'
+    echo 'make'
+    echo 'elif [[ "$1" == "install" ]]; then'
+    echo 'mkdir -p bin'
+    echo 'cd bin'
+    echo 'cmake ..'
+    echo 'sudo make install'
+    echo 'elif [[ "$1" == "debug" ]]; then'
+    echo 'mkdir -p debug'
+    echo 'cd debug'
+    echo 'cmake -DCMAKE_BUILD_TYPE=Debug ..'
+    echo 'make'
+    echo 'elif [[ "$1" == "project" ]]; then'
+    echo 'mkdir -p bin'
+    echo 'cd bin'
+    echo 'cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1 ..'
+    echo 'cp compile_commands.json ..'
+    echo 'fi'
+    ) > $project_name/build
+
+  chmod u+x $project_name/build
+
+  (
+    echo 'bin/*'
+    echo 'debug/*'
+    echo 'compile_commands.json'
+    echo '.vimspector.json'
+    echo '.clangd/*'
+  ) > $project_name/.gitignore
+
+  (
+    echo '{'
+    echo '"configurations": {'
+    echo '"Launch": {'
+    echo '"adapter": "vscode-cpptools",'
+    echo '"configuration": {'
+    echo '"request": "launch",'
+    echo "\"program\": \"debug/$exe_name\","
+    echo "\"cwd\": \"`pwd`\"," >> .vimspector.json
+    echo '"externalConsole": true,'
+    echo '"MIMode": "gdb"'
+    echo '}'
+    echo '}'
+    echo '}'
+    echo '}'
+  ) > $project_name/.vimspector.json
+
+  [ "$git_init" == "true" ] && git init $project_name
+
+  cd $project_name && ./build project
+
+}
+
+function py_init(){
+
+  [[  -z "$project_name" ]] && help && exit
+  echo $project_name
+  mkdir $project_name || exit
+  _project_name=$(echo $project_name | sed -e 's/-/_/g')
+  mkdir -p $project_name/docs
+  mkdir -p $project_name/$_project_name
+  mkdir -p $project_name/tests
+  touch $project_name/LICENSE
+  touch $project_name/README.md
+  touch $project_name/TODO.md
+  touch $project_name/setup.py
+  touch $project_name/.gitignore
+  touch $project_name/install.sh
+  touch $project_name/$_project_name/__init__.py
+  touch $project_name/$_project_name/utils.py
+  touch $project_name/$_project_name/__main__.py
+  [ "$git_init" == "true" ] && git init $project_name
+
+}
+
+function help(){
+  echo "usage: projectCreate -l lang -p projectName -c className"
+  echo "optional : -g intialise a git repo inside project folder"
+  echo "         : -h show this help message"
+}
```

Comment:
i was not able to retrieve this help function from command line. calling -h leads to some random help things. Can you check it out. Or I am missing something in here.

Reply:
I fixed that , I should have tested the script completely before opening this pr 😅

Reply:
No problem. This is fine. :smile: